core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -250,8 +250,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def raceAttempt[R1 <: R, E1 >: E, A1 >: A](that: ZIO[R1, E1, A1]): ZIO[R1, E1, A1] =
     raceWith(that)(
-      { case (l, f) => l.fold(f.interrupt *> ZIO.halt(_), ZIO.succeed) },
-      { case (r, f) => r.fold(f.interrupt *> ZIO.halt(_), ZIO.succeed) }
+      { case (l, f) => l.fold(f.interrupt *> ZIO.halt(_), f.interrupt *> ZIO.succeed(_)) },
```

Comment:
Small change for readability:
```suggestion
      { case (l, f) => f.interrupt *> l.fold(ZIO.halt, ZIO.succeed) },
```